base/docs/basedocs.jl

```diff
@@ -1977,57 +2006,146 @@ Boolean type, containing the values `true` and `false`.
+and [`significand`](@ref) to access various bits.
 
-        Binary format: $($sign) sign, $($exp) exponent, $($frac) fraction bits.
-        """
-        $(Symbol("Float", bit))
-    end
-end
+This is the default for floating point literals, `1.0 isa Float64`,
+and for many operations such as `1/2, 2pi, log(2), range(0,90,length=4)`.
+Unlike integers, this default does not change with `Sys.WORD_SIZE`.
+
+The exponent for scientific notation can be entered as `e` or `E`,
+thus `2e3 === 2.0E3 === 2.0 * 10^3`. This is strongly preferred over
+`10^n` because integers overflow, thus `10^19 < 0`.
+
+See also [`Inf`](@ref), [`NaN`](@ref), [`floatmax`](@ref), [`Float32`](@ref), [`Complex`](@ref).
+"""
+Float64
+
+"""
+    Float32 <: AbstractFloat <: Real
+
+32-bit floating point number type (IEEE 754 standard).
+Binary format is 1 sign, 8 exponent, 23 fraction bits.
+
+The exponent for scientific notation should be entered as lower-case `f`,
+thus `2f3 === 2.0f0 * 10^3 === Float32(2_000)`.
+For arrays literals and comprehensions, the element type can be specified before
+the square brackets: `Float32[1,4,9] == Float32[i^2 for i in 1:3]`.
+
+See also [`Inf32`](@ref), [`NaN32`](@ref), [`Float16`](@ref), [`exponent`](@ref), [`frexp`](@ref).
+"""
+Float32
+
+"""
+    Float16 <: AbstractFloat <: Real
+
+16-bit floating point number type (IEEE 754 standard).
+Binary format is 1 sign, 5 exponent, 10 fraction bits.
+"""
+Float16
 
 for bit in (8, 16, 32, 64, 128)
+    Sys.WORD_SIZE == bit && continue  # Int & UInt have separate, more detailed, descriptions
+    type = Symbol(:Int, bit)
+    unshow = repr(eval(Symbol(:UInt, bit))(bit-1))
+
     @eval begin
         """
-            Int$($bit) <: Signed
+            Int$($bit) <: Signed <: Integer
 
         $($bit)-bit signed integer type.
+
+        Note that integers overflow without warning,
+        thus `typemax($($type)) + $($type)(1) < 0`.
+        See also [`Int`](@ref $Int), [`widen`](@ref), [`BigInt`](@ref).
         """
         $(Symbol("Int", bit))
 
         """
-            UInt$($bit) <: Unsigned
+            UInt$($bit) <: Unsigned <: Integer
 
         $($bit)-bit unsigned integer type.
+
+        Printed in hexadecimal, thus $($(unshow)) == $($(bit-1)).
         """
         $(Symbol("UInt", bit))
     end
 end
 
+@eval begin
+    """
+        $Int === Int
+
+    $(Sys.WORD_SIZE)-bit signed integer type, `$Int <: Signed <: Integer <: Real`.
+
+    This is the default type of most integer literals, and has an alias `Int`.
+    It is the type returned by functions such as [`length`](@ref), and the standard type for indexing arrays.
+    The default (and the alias) may be 32 or 64 bits on different computers, indicated by `Sys.WORD_SIZE`.
+
+    Note that integers overflow without warning, thus `typemax(Int) + 1 < 0` and `10^19 < 0`.
+    Overflow can be avoided by using [`BigInt`](@ref), and is checked by most operations on [`Rational`](@ref).
```

Comment:
Most? I thought all rational operations were checked, but I'm not particularly confident.

Reply:
Likewise! Even if all in Base are, some package may define operations which aren't.

Reply:
On second thought, I don't think this statement about rationals belongs in the Int docstring. At most a "see also Rational" but even that seems like a bit of a stretch.

Reply:
Could be removed:
```suggestion
    Overflow can be avoided by using [`BigInt`](@ref).
```

Reply:
Done.